bofire/surrogates/mixed_single_task_gp.py

```diff
@@ -47,28 +51,29 @@ def _fit(self, X: pd.DataFrame, Y: pd.DataFrame):
             Y.values
         ).to(**tkwargs)
 
-        features2idx, _ = self.inputs._get_transform_info(
-            self.input_preprocessing_specs
+        continuous_feature_keys = get_continuous_feature_keys(
+            self.inputs, self.input_preprocessing_specs
+        )
+        ord_dims = self.inputs.get_feature_indices(
+            self.input_preprocessing_specs, continuous_feature_keys
         )
-        non_numerical_features = [
-            key
-            for key, value in self.input_preprocessing_specs.items()
-            if value != CategoricalEncodingEnum.DESCRIPTOR
-        ]
-
-        ord_dims = []
-        for feat in self.inputs.get():
-            if feat.key not in non_numerical_features:
-                ord_dims += features2idx[feat.key]
 
+        categorical_feature_keys = get_categorical_feature_keys(
+            self.input_preprocessing_specs
+        )
         # these are the categorical dimesions after applying the OneHotToNumeric transform
         cat_dims = list(
-            range(len(ord_dims), len(ord_dims) + len(non_numerical_features))
+            range(len(ord_dims), len(ord_dims) + len(categorical_feature_keys))
```

Comment:
`get_feature_indices` returns categorical dimensions with OneHot transformation applied, but we need the categorical dimensions without the transformation here.

Reply:
You are correct, I overlooked it. This is then also the position in the code, where we run into problems, when we have for example CategoricalDescriptorInputs of which we use some as OneHots and some as descriptors. This breaks then the indices, because we just rely here that the categorical features are always the last ones (here also the `order_id` comes into the play). This is often the case, but not always. And with the molecular ones coming in, it could be that it less often the case. But this bug, already existed before your PR. So we do not have to fix it here, but maybe you have a smart idea for the problem? ;)